src/sentry/tasks/app_store_connect.py

```diff
@@ -39,6 +39,18 @@ def inner_dsym_download(project_id: int, config_id: str) -> None:
 
     project = Project.objects.get(pk=project_id)
     config = appconnect.AppStoreConnectConfig.from_project_config(project, config_id)
+    project.update_option("sentry:last_checked_appstore", datetime.now())
```

Comment:
This is currently updated **before** the `client.list_builds()` call.  That is, we don't even know if we did anything at all.  I think instead we should update this option after we are done updating the database table, that way if there is a critical exception during the API requests and database updates we won't be updating the timestamp and that's probably the right behaviour.  So currently that would be line 66 i think